examples/md-flexible/src/Simulation.h

```diff
@@ -1,189 +1,303 @@
+     */
+    autopas::utils::Timer haloParticleExchange;
+
+    /**
+     * Records the time required to exchange migrating particles.
+     */
+    autopas::utils::Timer migratingParticleExchange;
+  };
 
   /**
-   * Gives an estimate of how many iterations the simulation will do in total.
-   * @return The estimate and true iff this is the correct number and not only an estimate.
+   * The timers used during the simulation.
    */
-  [[nodiscard]] std::tuple<size_t, bool> estimateNumIterations() const;
+  struct Timers _timers;
 
   /**
-   * Prints statistics like duration of calculation etc of the Simulation.
-   * @param autopas
+   * Parallel VTK file writer.
    */
-  void printStatistics(autopas::AutoPas<ParticleType> &autopas);
+  std::shared_ptr<ParallelVtkWriter> _vtkWriter;
 
   /**
-   * Getter for ParticlePropertiesLibrary of Simulation.
-   * @note Used for testing.
-   * @return unique_prt(ParticlePropertiesLibrary)
+   * Defines, if vtk files should be created or not.
    */
-  [[nodiscard]] const std::unique_ptr<ParticlePropertiesLibrary<double, size_t>> &getPpl() const;
+  bool _createVtkFiles;
 
  private:
   /**
-   * Print a progressbar and progress information to the console.
-   * @note Calling this function deletes the whole current line in the terminal.
-   *
-   * @param iterationProgress
-   * @param maxIterations
-   * @param maxIsPrecise Indicate whether maxIterations is precise (true) or an estimate (false).
+   * Estimates the number of tuning iterations which ocurred during the simulation so far.
+   * @return an estimation of the number of tuning iterations which occured so far.
+   */
+  std::tuple<size_t, bool> estimateNumberOfIterations() const;
+
+  /**
+   * Prints a progress bar to the terminal.
+   * @param iterationProgress: the number of already computed iterations.
+   * @param maxIterations: the number of maximum iterations.
+   * @param maxIsPrecise: Decides if the "~" symbol will be printed before the max iterations.
    */
   void printProgress(size_t iterationProgress, size_t maxIterations, bool maxIsPrecise);
 
   /**
-   * Use the variant of the LJFunctor that uses the shifted Lennard-Jones potential.
+   * Turns the timers into a human readable string.
+   * @param name: The timer's name.
+   * @param timeNS: The time in nano seconds.
+   * @param numberWidth: The precision of the printed number.
+   * @param maxTime: The simulation's total execution time.
+   * @return All information of the timer in a human readable string.
    */
-  constexpr static bool _shifting = true;
+  std::string timerToString(const std::string &name, long timeNS, size_t numberWidth = 0ul, long maxTime = 0ul);
+
   /**
-   * Use the variant of the LJFunctor that supports mixing of particle types.
+   * Calculate the homogeneity of the scenario by using the standard deviation.
+   * @return homogeneity
    */
-  constexpr static bool _mixing = true;
+  double calculateHomogeneity() const;
```

Comment:
This function as been moved to autopas::utils::... so it can be removed here, right?